sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -326,6 +327,25 @@ case class Join(
     hint: JoinHint)
   extends BinaryNode with PredicateHelper {
 
+  override def maxRows: Option[Long] = {
+    joinType match {
+      case Inner | Cross | FullOuter | LeftOuter | RightOuter
+        if left.maxRows.isDefined && right.maxRows.isDefined =>
+        val maxRows = BigInt(left.maxRows.get) * BigInt(right.maxRows.get)
+        if (maxRows.isValidLong) {
+          Some(maxRows.toLong)
+        } else {
+          None
```

Comment:
Could you add tests for this case in `test("SPARK-33497: Eliminate Limit if Join max rows not larger than Limit")`?

Reply:
Added by
```
  // maxRow is not valid long
  checkPlanAndMaxRow(
    testRelation.join(testRelation4, joinType).limit(100),
    testRelation.join(testRelation4, joinType).limit(100),
    100
  )
```